torch/_ops.py

```diff
@@ -128,6 +135,23 @@ def inner(fn):
 
         return inner
 
+    # Special registrar for ProxyTensorMode that handles basic stuff that
+    # every proxy tensor mode implementation needs to handle
+    def py_proxy_impl(self, fn):
+        from torch.fx.experimental.proxy_tensor import (
+            ProxyTorchDispatchMode,
+            set_original_aten_op,
+        )
+
+        def wrapped_proxy_fn(mode, *args, **kwargs):
+            with mode.sym_mode.enable(False), set_original_aten_op(self):
+                if not mode.enable_tracing:
+                    return self(*args, **kwargs)
+
+                return fn(mode, *args, **kwargs)
```

Comment:
This handles some common preamble that should be applied to all ProxyTorchDispatchMode registrations. The enable tracing is mostly not necessary as most HOPs handle it correctly internally, but if you wanted to, HOPs could delete this code now.

If people want to bikeshed the API, happy to. I need this to be a change that has NO user visible change as of this diff, because there are HOPs that are out of tree https://www.internalfb.com/code/fbsource/[550a8c8f5bba16c5a08a5ea885c085b435524366]/fbcode/executorch/exir/delegate.py?lines=20%2C112 https://www.internalfb.com/code/fbsource/[550a8c8f5bba16c5a08a5ea885c085b435524366]/fbcode/sigmoid/inference/passes/const_folding.py?lines=14%2C21

Reply:
In the future I'm curious if we can separate py_proxy_impl and py_impl to be separate and get py_proxy_impl to share code with https://github.com/pytorch/pytorch/blob/d224857b3af5c9d5a3c7a48401475c09d90db296/torch/fx/experimental/proxy_tensor.py#L546. There are other things that ProxyTensor does that we're not doing here (like constant prop)

Reply:
I kind of want to have an API that goes through the ProxyDispatchMode class rather than HigherOrderOperator, but that's a breaking API change so I have to hoover all the HOPs into pytorch first

Reply:
Going through ProxyDispatchMode seems reasonable